tests/test_generator.py

```diff
@@ -0,0 +1,361 @@
+    # Create Job
+    launch_settings = LaunchSettings(wlmutils.get_test_launcher())
+    app = Application("app_name", exe="python", run_settings="RunSettings")
+    job = Job(app, launch_settings)
+    # Mock start id
+    run_id = "mock_run"
+    # Generator instance
+    gen = Generator(exp_path=experiment_path, run_id=run_id)
+    # Call Generator.generate_job
+    job_path = gen.generate_job(job, 1)
+    assert isinstance(job_path, pathlib.Path)
+    expected_run_path = (
+        pathlib.Path(experiment_path)
+        / run_id
+        / f"{job.__class__.__name__.lower()}s"
+        / f"{app.name}-{1}"
+        / "run"
+    )
+    assert job_path == expected_run_path
+    expected_log_path = (
+        pathlib.Path(experiment_path)
+        / run_id
+        / f"{job.__class__.__name__.lower()}s"
+        / f"{app.name}-{1}"
+        / "log"
+    )
+    assert osp.isdir(expected_run_path)
+    assert osp.isdir(expected_log_path)
+    assert osp.isfile(osp.join(expected_log_path, "smartsim_params.txt"))
+
+
+def test_exp_private_generate_method_app(wlmutils, test_dir, generator_instance):
+    """Test that Job directory was created from Experiment."""
+    exp = Experiment(name="experiment_name", exp_path=test_dir)
+    app = Application("name", "python", "RunSettings")
+    launch_settings = LaunchSettings(wlmutils.get_test_launcher())
+    job = Job(app, launch_settings)
+    job_execution_path = exp._generate(generator_instance, job, 1)
+    assert osp.isdir(job_execution_path)
+    head, _ = os.path.split(job_execution_path)
+    expected_log_path = pathlib.Path(head) / "log"
+    assert osp.isdir(expected_log_path)
+
+
+def test_generate_copy_file(fileutils, wlmutils, test_dir):
+    # Create the Job and attach generator file
+    launch_settings = LaunchSettings(wlmutils.get_test_launcher())
+    app = Application("name", "python", "RunSettings")
+    script = fileutils.get_test_conf_path("sleep.py")
+    app.attach_generator_files(to_copy=script)
+    job = Job(app, launch_settings)
+
+    # Create the experiment
+    experiment_path = osp.join(test_dir, "experiment_name")
+    gen = Generator(exp_path=experiment_path, run_id="temp_run")
+    path = gen.generate_job(job, 1)
+    expected_file = pathlib.Path(path) / "sleep.py"
+    assert osp.isfile(expected_file)
+
+
+def test_generate_copy_directory(wlmutils, test_dir, get_gen_copy_file):
+    # Create the Job and attach generator file
+    launch_settings = LaunchSettings(wlmutils.get_test_launcher())
+    app = Application("name", "python", "RunSettings")
+    app.attach_generator_files(to_copy=get_gen_copy_file)
+    job = Job(app, launch_settings)
+
+    # Create the experiment
+    experiment_path = osp.join(test_dir, "experiment_name")
+    gen = Generator(exp_path=experiment_path, run_id="temp_run")
+    path = gen.generate_job(job, 1)
+    expected_file = pathlib.Path(path) / "mock.txt"
+    assert osp.isfile(expected_file)
```

Comment:
Not wrong, just sanity checking: it looks like this test is testing that when copying a dir, the contents of the dir are copied, NOT the dir itself.

e.g. this test generates
```sh
{test_dir}/experiment_name/temp_run/jobs/name-1/run/mock.txt
```
NOT
```sh
{test_dir}/experiment_name/temp_run/jobs/name-1/run/to_copy_dir/mock.txt
#                                                   ^^^^^^^^^^^
# This is not present in the previous path
```

Is this intended? I know it might be hold over from V0.7 because smartsim did some weird things with copying directories in the past, but I might ask that we change it, especially since its inconsistent with `to_symlink`. Probably worth getting group insight on this though!